### PR TITLE
feature(coq): coq macro

### PR DESCRIPTION
**Pull request by Alizter**
This is a continuation of #5967. The difference here is that the extra variables have been removed in a following commit. We now only expose:

Coq configuration variables:
```ocaml
%{coq:coqlib}
%{coq:coq_native_compiler_default}
```

Version numbers for Coq:
```ocaml
%{coq:version}
%{coq:ocaml-version}
%{coq:version.major}
%{coq:version.minor}
%{coq:version.suffix}
```

- close https://github.com/ocaml/dune/pull/5967
- close https://github.com/ocaml/dune/pull/5913

### CODE REVIEW

**Comment by Alizter** [comment]
@ejgallego ping

**Comment by ejgallego** [comment]
Looks very good to me, thanks! I can do a full code review next week after I'm done with the Sabanci class, but if someone else wants to take over, the interface for this PR looks good to me so you can merge.

Need to update the docs and changelog I think.

**Comment by ejgallego** [comment]
Also `%{coq:coq_native_compiler_default}` could be maybe named a bit differently? The flag really means that `coqc` was configured to output native files by default. But still you can use the native compiler when this flag in configure was not passed.

It is a hack so people could have coq_makefile output native files just by recompiling coq.

**Comment by Alizter** [comment]
> Also `%{coq:coq_native_compiler_default}` could be maybe named a bit differently? The flag really means that `coqc` was configured to output native files by default. But still you can use the native compiler when this flag in configure was not passed.
> 
> It is a hack so people could have coq_makefile output native files just by recompiling coq.

We can name it whatever we want, I just kept the env var name from configure.

**Comment by Alizter** [comment]
@ejgallego ping

File: test/blackbox-tests/test-cases/coq/coq-config.t/dune
Comment (ejgallego) [inline test/blackbox-tests/test-cases/coq/coq-config.t/dune:27]: Maybe you want to use grep to filter out the positive ones instead (use the egrep `|` pattern for disjunctive regexps)

**Review by ejgallego — APPROVED** [review]
Looks very good to me but I'm not expert on the expander code, so it'd be nice if an expert could have a look, maybe (@rgrinberg ) ?

**Review by ejgallego — CHANGES_REQUESTED** [review]
Actually CHANGES and doc need updating.

File: test/blackbox-tests/test-cases/coq/coq-config.t/dune
Comment (Alizter) [reply test/blackbox-tests/test-cases/coq/coq-config.t/dune:27]: Grep might not run the same in mac so I stuck to sed

**Comment by Alizter** [comment]
@ejgallego I've added a changelog and some doc.

**Comment by Alizter** [comment]
The expander code is originally @rgrinberg's, but if he has some time, I would mind another pair of eyes.

File: CHANGES.md
Comment (christinerose) [inline CHANGES.md:48]:
```suggestion
- odoc rules now about `ODOC_SYNTAX` and will rerun accordingly (#6010, fixes
```
I know it looks strange, but rerun is written without the hyphen.

File: doc/coq.rst
Comment (christinerose) [inline doc/coq.rst:549]:
```suggestion
Coq-Specific Variables
```

File: doc/coq.rst
Comment (christinerose) [inline doc/coq.rst:556]:
```suggestion
- ``%{coq:version.major}`` the major version of Coq (e.g., ``8.15.2`` gives
```

File: doc/coq.rst
Comment (christinerose) [inline doc/coq.rst:558]:
```suggestion
- ``%{coq:version.minor}`` the minor version of Coq (e.g., ``8.15.2`` gives
```

File: doc/coq.rst
Comment (christinerose) [inline doc/coq.rst:560]:
```suggestion
- ``%{coq:version.suffix}`` the suffix version of Coq (e.g., ``8.15.2`` gives
```

File: test/blackbox-tests/test-cases/coq/coq-config.t/run.t
Comment (christinerose) [inline test/blackbox-tests/test-cases/coq/coq-config.t/run.t:1]:
```suggestion
We test the Coq macro
```
Perhaps "Then test the Coq macro" or "To test the Coq marcro" ?

**Review by christinerose — APPROVED** [review]
Mostly minor grammar/formatting suggestions. One question at the end about phrasing.

File: CHANGES.md
Comment (Alizter) [reply CHANGES.md:48]: Let's fix this in another PR since I didn't introduce that here.

Comment (Alizter) [reply CHANGES.md:48]: I've opened #6155.

File: test/blackbox-tests/test-cases/coq/coq-config.t/run.t
Comment (Alizter) [reply test/blackbox-tests/test-cases/coq/coq-config.t/run.t]: I've changed it to "Testing the Coq macro". It doesn't have to be more informative than that.

File: CHANGES.md
Comment (christinerose) [reply CHANGES.md:53]: Thanks! Approved that one.

File: CHANGES.md
Comment (ejgallego) [inline CHANGES.md:56]: Is the double about here needed?

**Review by ejgallego — APPROVED** [review]
Great, will merge once the conflict is solved.

File: CHANGES.md
Comment (Alizter) [reply CHANGES.md:56]: Doesn't seem to be the case in the changelog.

**Pull request merged**
